### ISSUE TITLE
[META] Optimize integrations use of @metadata._id in all integrations using it

### ISSUE THREAD

**Issue by lucabelluccini** [elastic/integrations#5100]
We are aware the following integrations make use of `@metadata._id`.

**Some** of the Beats modules or Agent integrations forcing an explicit `@metadata._id` to avoid duplicates
- List of Elastic Agent integrations forcing an `_id`: https://github.com/search?q=repo%3Aelastic%2Fintegrations+%22target_field%3A+_id%22+path%3Apackages%2F&type=code
- List of Elastic Agent integrations and Beat Inputs using explicit `_id`: https://github.com/search?q=repo%3Aelastic%2Fbeats%20%22event.SetID%22%20path%3Ainput%2F&type=code
- List of Beats modules using explicit `_id` (on top of inputs): https://github.com/search?q=repo%3Aelastic%2Fbeats%20%22processor.Fingerprint%22&type=code

The use of explicit `_id` can lead to slower indexing performances on Elasticsearch due to the costs of lookups.

Where **applicable**, the modules/integrations should prefer using the following technique to generate `_id`.

If explicit identifiers are required to avoid event duplication, prefer identifiers prefixed by a timestamp, which are much more efficient than random identifier.
This [blog](https://www.elastic.co/blog/efficient-duplicate-prevention-for-event-based-data-in-elasticsearch) explains in details why.
In short, identifier prefixed by a timestamp have two benefits:
- they gradually increase over time based on sorting order, so Elasticsearch can quickly determine whether an identifier exists in older index segments based solely on the minimum and maximum identifier value in that segment rather than having to search through it
- they reduce the hash collision probability, as the hash value only has to be unique per timestamp

This issue would involve all integration owners/maintainers.
